src/main/scala/org/apache/spark/sql/rikai/NDArray.scala

```diff
@@ -17,10 +17,46 @@
 package org.apache.spark.sql.rikai
 
 import org.apache.spark.sql.types._
+import org.apache.spark.sql.catalyst.expressions.GenericInternalRow
+import org.apache.spark.sql.catalyst.InternalRow
 
+/**
+  * NDArray type. Storing the persisted data from numpy ndarray.
+  *
+  * Note that this class does not support interoperation in scala yet.
+  * It will only show a array summary in "df.show()".
+  * But the data is accessible from Pyspark and Pytorch / Tensorflow readers.
+  *
+  * @todo consider to use arrow Array for interchangable data transfer.
```

Comment:
I don't thnk we should invest more in scala interop for numpy ndarray. Let's wait til we're able to take advantage of more Arrow integrations.

Reply:
agreed. so put this into `todo` :)